### ISSUE TITLE
Responsiveness Issue with small screen width

### ISSUE THREAD

**Issue by faizan-carameltech** [react-syntax-highlighter/react-syntax-highlighter#219]
**Responsiveness Issue**
The line number and code string overlap when a small width is active.

**To Reproduce**
Simply drag the browser window to minimize the screen width to a minimum

**Code Sample**
```javascript
const codeString = `// initialize fizz
IFizzClient fizzClient = IFizzClient::instance();
fizzClient.init("YOUR-APP-ID", "YOUR-APP-Secret",
  "writablePath to keep session logs required by fizz c++ client");

// add listener
fizzClient.chat()->listener()->addOnMessagePublishedListener([=](FizzChannelMessagePtr message) {
  // messages received will be notified here
}, "Your Custom Key for Listener");

// open connection and send message
fizzClient.open("userId", FizzLanguageCodes.English, FizzServices.All, [=](IFizzErrorPtr openError) {
  std::map<std::string, std::string> dataMap;
  bool translate = true, persist = true, filter = true;

  fizzClient.chat()->publishMessage("channelId", "userNick", "msgText",
    dataMap, translate, persist, filter, [=](IFizzErrorPtr publishError) {
  });
});`
```
```html
<SyntaxHighlighter
        style={xcode}
        language={'cpp'}
        showLineNumbers={true}
        customStyle={{
                      margin: 0,
                      wordBreak: 'break-all',
                      whiteSpace: 'pre-wrap',
                      boxShadow: '0px 2px 4px rgba(50,50,93,.1)'
        }}
        lineNumberStyle={{
                      color: '#5dc79e',
                      textAlign: 'right',
                      marginLeft: '-0.5em',
                      background: '#f2feef',
                      padding: '0 10px 0 10px'
        }}
>
                    {codeString}
</SyntaxHighlighter>
```

**Current behavior**
<img width="443" alt="reactSyntax" src="https://user-images.githubusercontent.com/18396012/66558138-83199d00-eb6c-11e9-9968-033d854edd7d.png">

**Expected behavior**
<img width="464" alt="runKit" src="https://user-images.githubusercontent.com/18396012/66558178-9298e600-eb6c-11e9-8574-7c9884784cd7.png">

**Comment by faizan-carameltech** [comment]
@conorhastings, help needed. TIA

**Comment by conorhastings** [comment]
@faizan-carameltech I don't currently have the bandwidth to look into this issue, I would reccomend exploring other ways to use line numbers manually as this is brittle and will likely be removed

**Comment by patmigliaccio** [comment]
See the following solution, may lend to what could to be updated in the internals.

```html
<SyntaxHighlighter lineProps={{className: 'code-line'}}>
  {codeString}
</SyntaxHighlighter>
```

```css
  code {
    counter-reset: line;
  }

  .code-line {
    counter-increment: line;
    position: relative;
    display: block;
    margin-left: 1.5rem;
  }

  .code-line:before {
    content: counter(line);
    position: absolute;
    margin-left: -1.5rem;
  }
```

It ignores the `showLineNumbers` prop and sets them via CSS instead of rendering elements in the DOM.

If I find some time this week, will try to see if I can implement.

**Comment by jaredpetersen** [comment]
@patmigliaccio This is an amazing fix! Thank you!

**Comment by conorhastings** [comment]
@patmigliaccio I don't really fully understand what is happening there so you find time to implement that would be great as i don't feel confident doing it

**Comment by simmerer** [comment]
This is fixed by adding the `showInlineLineNumbers` prop, available as of `react-syntax-highlighter 13.0.0`.

```javascript
<SyntaxHighlighter
        style={xcode}
        language={'cpp'}
        showLineNumbers={true}
        showInlineLineNumbers={true} // <-- add this prop!
        customStyle={{
                      margin: 0,
                      wordBreak: 'break-all',
                      whiteSpace: 'pre-wrap',
                      boxShadow: '0px 2px 4px rgba(50,50,93,.1)'
        }}
        lineNumberStyle={{
                      color: '#5dc79e',
                      textAlign: 'right',
                      marginLeft: '-0.5em',
                      background: '#f2feef',
                      padding: '0 10px 0 10px'
        }}
>
                    {codeString}
</SyntaxHighlighter>
```
